ios-xcframework/Podfile

```diff
@@ -9,57 +9,40 @@ HERMES_ENABLED = ENV.fetch('HERMES_ENABLED', true)
 
 puts "[Gutenberg] Installing pods with Hermes #{HERMES_ENABLED ? 'enabled' : 'disabled'}"
 
-# Note that the pods in this array might seem unused if you look for
-# `import` statements in this codebase. However, make sure to also check
-# whether they are used in the gutenberg-mobile and Gutenberg projects.
-#
-# Also notice that these are not all the dependencies the project uses.
-# Later in the config, we call use_react_native! which fetches more.
-#
-# See https://github.com/wordpress-mobile/gutenberg-mobile/issues/5025
-DEPENDENCIES = %w[
-  react-native-safe-area
-  react-native-safe-area-context
-  react-native-video
-  react-native-webview
-  RNSVG
-  react-native-slider
-  BVLinearGradient
-  react-native-get-random-values
-  react-native-blur
-  RNScreens
-  RNReanimated
-  RNGestureHandler
-  RNCMaskedView
-  RNCClipboard
-  RNFastImage
-  React-jsc
-].freeze
-
-def gutenberg_dependencies
-  podspec_prefix = '..'
-
-  computed_dependencies = DEPENDENCIES.dup
-
-  # Use a custom RNReanimated version while we coordinate a fix upstream
-  computed_dependencies.delete('RNReanimated')
-
-  computed_dependencies.delete('React-jsc') unless HERMES_ENABLED
-
-  computed_dependencies.each do |pod_name|
-    pod pod_name, podspec: "#{podspec_prefix}/third-party-podspecs/#{pod_name}.podspec.json"
-  end
-
+def setup_rnreanimated_pod_from_fork
   # This is required to workaround an issue with RNReanimated after upgrading to version 2.17.0
   rn_node_modules = File.join(Dir.pwd, '..', 'gutenberg', 'node_modules')
   raise "Could not find node modules at given path #{rn_node_modules}" unless File.exist? rn_node_modules
+
   ENV['REACT_NATIVE_NODE_MODULES_DIR'] = rn_node_modules
   puts "[Gutenberg] Set REACT_NATIVE_NODE_MODULES_DIR env var for RNReanimated to #{rn_node_modules}"
 
   # Use a custom RNReanimated version while we coordinate a fix upstream
   pod 'RNReanimated', git: 'https://github.com/wordpress-mobile/react-native-reanimated', branch: 'wp-fork-2.17.0'
 end
 
+def gutenberg_dependencies
+  podspec_extension = '.podspec.json'
+
+  Dir["../third-party-podspecs/*#{podspec_extension}"].each do |podspec_path|
+    pod_name = File.basename(podspec_path, podspec_extension)
```

Comment:
It might be more robust to use the podspec json's `name` property as the `pod_name`, instead of the filename? We probably won't use a different name for the podspec.json file intentionally, but there might be a typo, omitting a dash (i.e. RCTFolly.podspec.json for the RCT-Folly pod), or something else, which would cause a mismatch between the filename and the pod name.